docs/advanced-usage/external-iscsi.md

```diff
@@ -0,0 +1,170 @@
+    ```
+K8s 会调用对应 IaaS platform 的接口为你创建的 service 分配一个 external-ip ，作为 LoadBalancer 的入口。
+
+最后，执行 `kubectl edit iomesh -n iomesh-system`，将 `spec.redirector.iscsiVirtualIP` 设置为 `iomesh-access-lb` service 的 external-ip，保存后 `iomesh-iscsi-redirector` Pod 会自动重启生效设置。
+
+<!--Out-Tree LoadBalancer-->
+1. 创建文件 `iomesh-access-lb-service.yaml` 并写入如下内容:
+
+    ```yaml
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: iomesh-access-lb
+      namespace: iomesh-system
+    spec:
+      ports:
+      - name: iscsi
+        port: 3260
+        protocol: TCP
+        targetPort: 3260
+      selector:
+        app.kubernetes.io/name: iomesh-iscsi-redirector
+      type: LoadBalancer
+    ```
+
+2. 应用 service 配置
+
+    ```bash
+    kubectl apply -f iomesh-access-lb-service.yaml
+    ```
+
+3. 检查 service 状态
+
+    ```bash
+    kubectl get service iomesh-access-lb -n iomesh-system
+    ```
+
+由于 IOMesh 运行在 bare metal 环境，或其他非 K8s 原生支持 LoadBalancer type service 的 cloud provider 中，K8s 不提供 load balancer 的默认实现，创建的 LoadBalancer service 将一直处于 pending 状态。此时需要在 K8s 集群中安装 `MetalLB` 来作为默认的 load balancer 实现，安装和配置方式如下
+1. 确保集群满足 `MetalLB` 的[前置安装条件](https://metallb.universe.tf/installation/#preparation)
+
+2. 安装 `MetalLB`
+
+    ```bash
+    helm repo add metallb https://metallb.github.io/metallb
+    helm install metallb metallb/metallb --version 0.12.1
+    ```
+
+3. 创建 `MetalLB` ConfigMap 文件 `metallb-config.yaml`，设置 `MetalLB` 的工作模式为 layer2 (基于 arp)， 并为 `MetalLB` 分配 ip 池
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      namespace: iomesh-system
+      name: metallb
+    data:
+      config: |
+        address-pools:
+        - name: iomesh-access-address-pools
+          protocol: layer2
+          addresses:
+          - <fill-in-your-ip-address-pool-here> # eg: "192.168.1.100-192.168.1.110" or "192.168.2.0/24"
+    ```
+将 `<fill-in-your-ip-address-pool-here>` 部分替换为物理网络中未被使用的 ip 池，可以使用 ip range 的形式，如 "192.168.1.100-192.168.1.110"，也可以使用子网掩码的形式，如 "192.168.2.0/24"
+
+4. 应用 ConfigMap 配置
+
+    ```bash
+    kubectl apply -f metallb-config.yaml
+    ```
+
+5. 检查 service 状态，确保 `MetalLB` 分配了 ip 池中的 external-ip
+
+    ```bash
+    watch kubectl get service iomesh-access-lb -n iomesh-system
+    ```
+
+<br>
+最后，执行 `kubectl edit iomesh -n iomesh-system`，将 `spec.redirector.iscsiVirtualIP` 设置为 `iomesh-access-lb` service 的 external-ip，保存后 `iomesh-iscsi-redirector` Pod 会自动重启生效设置。
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## 使用 PVC 创建 external iSCSI LUN
+使用 PVC 创建 external iSCSI LUN 需要在 PVC 中增加两个 annotation 字段， 字段说明和完整的例子如下
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: external-iscsi
+  annotations:
+    # 标记该 PVC 对应一个集群外部使用的 LUN
+    iomesh.com/external-use: "true"
+    # 设置 LUN 的 initiator iqn acl，不配置该字段代表禁止所有 initiator login。PVC 的 accessModes 为 RWO 时该字段的值有
+    # 且只能有一条，PVC 的 accessModes 为 RWX 时该字段的值可以有多条，用逗号隔开。放行所有 iqn 配置为 "*/*"
+    iomesh.com/iscsi-lun-iqn-allow-list: "iqn.1994-05.com.redhat:a6c97f775dcb"
```

Comment:
iqn.1994-05.com.***redhat***:a6c97f775dcb  -> better to avoid using redhat branding here?